packages/api/pages/api/languages/[code]/import.ts

```diff
@@ -14,15 +18,35 @@ export default createRoute()
       import: z.string(),
     }),
     async handler(req, res) {
-      // TODO: implement import logic.
-      console.log('import language:', req.body.import);
-      // simulate the processing time
-      await new Promise<void>((resolve) => {
-        setTimeout(() => {
-          resolve();
-        }, 10000);
-      });
+      // TODO: we only use a few book keys for testing, to avoid creating a
+      //       ton of requests. use all keys when things are working
+      // for (const key of bookKeys) {
+      for (const key of ['Gen', '1Ch']) {
+        const importUrl = `${importServer}/${req.body.import}Glosses/${key}Gloss.js`;
+        const response = await fetch(importUrl);
+        const jsCode = await response.text();
+        const glossArray = parseGlossJs(jsCode);
+        console.log(glossArray.length);
+      }
       res.ok();
     },
   })
   .build();
+
+/**
+ * Parse JSON data from the JS gloss scripts.
+ */
+function parseGlossJs(jsCode: string) {
```

Comment:
Over commented this function on purpose! It uses a lot of regular expressions, and regex code is usually NOT very intuitive to read.